app/controllers/locations_controller.rb

```diff
@@ -1,40 +1,8 @@
 class LocationsController < ApplicationController
   def index
-    authorize Location
-    scope = Location.current.where(hidden: hidden_flags)
+    expires_in Rails.application.config.cache_max_age, public: true
 
-    @locations, @sorting_params = policy_scope(scope).alpha_paginate(
-      params[:letter],
-      ALPHABETICAL_PAGINATE_CONFIG.dup,
-      &:title
-    )
-  end
-
-  def update
-    location = Location.find(params[:id])
-    authorize location
-    if location.update_attributes(permitted_attributes(location))
-      flash[:notice] = "Successfully updated #{location.title}"
-    else
-      flash[:error] = "Error updating #{location.title}"
-    end
-    redirect_to_locations_directory(location)
-  end
-
-  private
-
-  def redirect_to_locations_directory(location)
-    redirect_to locations_path(
-      letter: location.title[0],
-      display_hidden: params[:display_hidden],
-      display_active: params[:display_active]
-    )
-  end
-
-  def hidden_flags
-    flags = []
-    flags << true if params[:display_hidden].present?
-    flags << false if params.fetch(:display_active, true).present?
-    flags
+    @locations = Location.where(state: 'current', hidden: false)
+    render :index
```

Comment:
Also, maybe a little nit-picky, but to be properly semantic we should probably provide a `respond_to` that only specifies `format.json` here.